internal/declarative/v2/reconciler.go

```diff
@@ -144,7 +144,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			r.cleanUpMandatoryModuleMetrics(obj)
 			return r.removeFinalizers(ctx, obj, []string{r.Finalizer}, metrics.ManifestRemoveFinalizerWhenParseSpec)
```

Comment:
I think here we're also ignoring the original error. It is clearly visible in the `removeFinalizers` function, in the line 256.